pkg/rbd/nodeserver.go

```diff
@@ -94,10 +97,13 @@ func (ns *NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 	}
 
 	stagingParentPath := req.GetStagingTargetPath()
-	stagingTargetPath := stagingParentPath + "/" + req.GetVolumeId()
+	stagingTargetPath := stagingParentPath + "/" + volID
 
-	idLk := nodeVolumeIDLocker.Lock(volID)
-	defer nodeVolumeIDLocker.Unlock(idLk, volID)
+	if acquired := ns.VolumeLocks.TryAcquire(volID); !acquired {
```

Comment:
Move this below line 75 above, fail fast. Previously we did work till we had no conflict with the prior request, now that we are aborting, there is little reason to do any work and abort.